### PR TITLE
Add combo, auto_shift to info_config and info_rule json

### PR DESCRIPTION
**Pull request by adnanaq**
## Description

Added Combo and Auto_shift related config into info_config, info_rule, and updated ijsonschema with related changes.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by adnanaq** [comment]
@zvecr I see you added `need doc `tag. What documentation do I need to add and where?

**Comment by zvecr** [comment]
It would be `docs/reference_info_json.md`. However as these are keymap level config, there is potentially going to need to be some rework before this gets accepted.

**Comment by adnanaq** [comment]
What kind of rework are we talking about? Is that something which can be done in the PR or there would be need for another PR?

File: data/schemas/keyboard.jsonschema [hunk 1]
```diff
@@ -150,6 +150,20 @@
                 "voices": {"type": "boolean"}
             }
         },
+        "auto_shift": {
+            "type": "object",
+            "properties": {
+                "enter": {"type": "boolean"},
```
Comment (drashna) [inline data/schemas/keyboard.jsonschema:156]: I kinda feel like all of the key configurations (eg, what it triggers and doesn't) should be in some subkey.  Not sure if that's the best way though

File: data/schemas/keyboard.jsonschema [hunk 2]
```diff
@@ -276,8 +290,12 @@
         "combo": {
             "type": "object",
             "properties": {
+                "enabled": {"type": "boolean"},
                 "count": {"$ref": "qmk.definitions.v1#/unsigned_int"},
-                "term": {"$ref": "qmk.definitions.v1#/unsigned_int"}
+                "term": {"$ref": "qmk.definitions.v1#/unsigned_int"},
+                "extra_short_combos": {"$ref": "qmk.definitions.v1#/unsigned_int"},
```
Comment (drashna) [inline data/schemas/keyboard.jsonschema:296]: same with this. Also, this feels like it should be "length" with short, extra short, extra long, etc as options.

File: data/schemas/keyboard.jsonschema (hunk 1 above)
Comment (adnanaq) [reply data/schemas/keyboard.jsonschema:156]: I was of same thought. Initially I added all keys prefixed with `no_` under `exclusion` subkey. So it is obvious that those keys wouldn't trigger auto shift behavior. I also thought of using `behavior` as a key but it's pretty broad term

File: data/schemas/keyboard.jsonschema (hunk 2 above)
Comment (adnanaq) [reply data/schemas/keyboard.jsonschema:296]: Yeah, "length" seems like more accurate subkey for thrse.

File: data/schemas/keyboard.jsonschema (hunk 1 above)
Comment (adnanaq) [reply data/schemas/keyboard.jsonschema:156]: I have updated PR with some changes and subkeys. Let me know if they make any sense. Or If there is a convention we'd be following.

**Pull request closed without merging**